corehq/form_processor/submission_post.py

```diff
@@ -449,30 +449,27 @@ def do_post_save_actions(case_db, xforms, case_stock_result):
             raise PostSaveError
 
     @staticmethod
-    def send_to_elasticsearch(instance):
+    def send_to_elasticsearch(instance, case_stock_result):
         if instance.metadata.deviceID != FORMPLAYER_DEVICE_ID:
             return
 
         domain_obj = Domain.get_by_name(instance.domain)
         if not domain_obj or not domain_obj.web_apps_sync_case_search:
             return
 
-        from casexml.apps.case.xform import extract_case_blocks
         from corehq.elastic import send_to_elasticsearch
         from corehq.pillows.case_search import transform_case_for_elasticsearch
-        case_ids = [info.get('@case_id') for info in extract_case_blocks(instance)]
-        cases = CommCareCase.objects.get_cases(case_ids)
-        for instance_case in cases:
+        for case_model in case_stock_result.case_models:
```

Comment:
nit: I'd pass in `case_models` directly since it's the only thing this method needs, that way it doesn't have to know what a `case_stock_result` is.

Reply:
2205e665390